pkg/scalers/metrics_api_scaler.go

```diff
@@ -172,39 +172,45 @@ func parseMetricsAPIMetadata(config *ScalerConfig) (*metricsAPIScalerMetadata, e
 }
 
 // GetValueFromResponse uses provided valueLocation to access the numeric value in provided body
-func GetValueFromResponse(body []byte, valueLocation string) (int64, error) {
+func GetValueFromResponse(body []byte, valueLocation string) (*resource.Quantity, error) {
 	r := gjson.GetBytes(body, valueLocation)
+	if r.Type == gjson.String {
+		q, err := resource.ParseQuantity(r.String())
+		if err == nil {
```

Comment:
What will be returned if the `err!=nil`?

Reply:
It would revert to the original behaviour and return "valueLocation must point to value of type number got: string"

Reply:
I've improved the error handling